### PR TITLE
Updated Native Models 

### PR DESCRIPTION
**Pull request by MadhuNimmo**
Added stubs/models to support more native method calls.

### CODE REVIEW

File: cast/js/src/main/resources/preamble.js [hunk 1]
```diff
@@ -133,21 +133,26 @@ DOMNode = function DOMNode() { // An impostor for the Node class
         
         this.selectNodes = function(a) {
 	}
+
+	this.hasAttribute = function(a) {
+	}
 };
 
-DOMNode.prototype.addEventListener = function Node_prototype_addEventListener(name, fn) { fn(); };
+DOMNode.prototype.addEventListener = function Node_prototype_addEventListener(name, fn, options) { fn(); };
```
Comment (msridhar) [inline cast/js/src/main/resources/preamble.js:141]: Why do we need this change?  Is some call edge missing if we don't do it?  It's an ok change, just wondering why it's needed

File: cast/js/src/main/resources/preamble.js [hunk 2]
```diff
@@ -133,21 +133,26 @@ DOMNode = function DOMNode() { // An impostor for the Node class
...
 
-DOMNode.prototype.removeEventListener = function Node_prototype_removeEventListener(name) {};
+DOMNode.prototype.removeEventListener = function Node_prototype_removeEventListener(name, fn, options) {};
```
Comment (msridhar) [inline cast/js/src/main/resources/preamble.js:143]: Same question as previous comment.  In general there are several changes that just fix the number of parameters to be passed in, which are fine, but I'd like to understand how they impact constructed call graphs

File: cast/js/src/main/resources/preamble.js [hunk 3]
```diff
@@ -646,4 +706,55 @@ function ActiveXObject() {
...
+	},
+	this.removeItem = function localStorage_removeItem(keyName){
+		// TODO: model me
+	},
+	this.clear = function localStorage_clear(){
+		// TODO: model me
+	}
+}
+
+Console = function Console(){
```
Comment (msridhar) [inline cast/js/src/main/resources/preamble.js:750]: Does this help with calls like `console.log(...)`?  If so, how, given that the variable is named `Console`?

File: cast/js/src/main/resources/prologue.js [hunk 4]
```diff
@@ -109,6 +150,8 @@ Object.prototype = Object$proto$__WALA__;
 /* Function properties, see spec 15.3					*/
 /************************************************************************/
 
+//Function = function Function() {};
+
```
Comment (msridhar) [inline cast/js/src/main/resources/prologue.js:154]: Why this change?

File: cast/js/src/main/resources/prologue.js [hunk 5]
```diff
@@ -532,6 +592,8 @@ local_string.prototype = String$proto$__WALA__;
 /* Number properties, see spec 15.7					*/
 /************************************************************************/
 
+//Number = function Number() {};
+
```
Comment (msridhar) [inline cast/js/src/main/resources/prologue.js:596]: Why this change?

**Review by msridhar — COMMENTED** [review]
Looks great!  I have a few questions

File: cast/js/src/main/resources/preamble.js (hunk 1 above)
Comment (MadhuNimmo) [reply cast/js/src/main/resources/preamble.js:141]: These changes don't impact the call edge. As I was updating models, I updated the signatures for some.

File: cast/js/src/main/resources/preamble.js (hunk 2 above)
Comment (MadhuNimmo) [reply cast/js/src/main/resources/preamble.js:143]: No impact on edges

File: cast/js/src/main/resources/preamble.js (hunk 3 above)
Comment (MadhuNimmo) [reply cast/js/src/main/resources/preamble.js:750]: Yes. Sample Program:
```

function getRandomNumber(min, max) {
        return Math.floor(Math.random() * (max - min + 1)) + min;
}

var randomNumber = getRandomNumber(1, 100);

console.log("Random Number:", randomNumber);

if (randomNumber % 2 === 0) {
        console.debug("The random number is even.");
} else {
        console.info("The random number is odd.");
}
    
```
 Output: 
 
```
   "test.js@1:0-341": {
    "test.js@12:297-338": [
      "console_info (Native)"
    ],
    "test.js@5:125-148": [
      "test.js@1:0-104"
    ],
    "test.js@7:151-194": [
      "console_log (Native)",
      "Math_log (Native)"
    ],
    "test.js@10:235-278": [
      "console_debug (Native)"
    ]
  }
```

File: cast/js/src/main/resources/prologue.js (hunk 4 above)
Comment (MadhuNimmo) [reply cast/js/src/main/resources/prologue.js:154]: These were to identify the constructor calls. But, some of the tests fail as of now, due to these constructor calls. I can remove them for now.

File: cast/js/src/main/resources/prologue.js (hunk 5 above)
Comment (MadhuNimmo) [reply cast/js/src/main/resources/prologue.js:596]: Same as Function.

File: cast/js/src/main/resources/preamble.js (hunk 3 above)
Comment (msridhar) [reply cast/js/src/main/resources/preamble.js:750]: Thanks, but can you explain why adding a write to the `Console` variable leads to new information for a variable named `console`?  These are two different variables.

File: cast/js/src/main/resources/prologue.js (hunk 4 above)
Comment (msridhar) [reply cast/js/src/main/resources/prologue.js:154]: Remove the changes?  Or remove the tests?

Comment (MadhuNimmo) [reply cast/js/src/main/resources/prologue.js:154]: Remove the changes

Comment (msridhar) [reply cast/js/src/main/resources/prologue.js:162]: Were you planning to remove this change?

File: cast/js/src/main/resources/prologue.js (hunk 5 above)
Comment (msridhar) [reply cast/js/src/main/resources/prologue.js:604]: Same question as above

File: cast/js/src/main/resources/prologue.js (hunk 4 above)
Comment (MadhuNimmo) [reply cast/js/src/main/resources/prologue.js]: done

File: cast/js/src/main/resources/prologue.js [hunk 6]
```diff
@@ -502,7 +555,12 @@ String$proto$__WALA__ = {
 
   replace: function String_prototype_replace(regex, withStr) {
     // return new String(primitive("StringReplace", this, regex, withStr));
-    return this || withStr;
+    //return this || withStr;
```
Comment (msridhar) [inline cast/js/src/main/resources/prologue.js:558]: Can we remove this commented line?

File: cast/js/src/main/resources/prologue.js [hunk 7]
```diff
@@ -744,4 +810,82 @@ function EvalError(str) {
...
+
+  then: function Promise_prototype_then (onFulfilled, onRejected) {
+    // TO DO
+  },
+  catch: function Promise_prototype_catch (onRejected) {
+    // TO DO
+  }
+}
 
+local_promise.prototype = Promise$proto$__WALA__;
```
Comment (msridhar) [inline cast/js/src/main/resources/prologue.js:891]: Can we add a newline at the end of the file?

**Review by msridhar — COMMENTED** [review]
Looks great!  Couple more very minor things and then this will be ready to land

File: cast/js/src/main/resources/prologue.js (hunk 6 above)
Comment (MadhuNimmo) [reply cast/js/src/main/resources/prologue.js]: done

File: cast/js/src/main/resources/prologue.js (hunk 7 above)
Comment (MadhuNimmo) [reply cast/js/src/main/resources/prologue.js]: done

**Review by msridhar — APPROVED** [review]
